src/rpy_diffusive_flux.cc

```diff
@@ -0,0 +1,166 @@
+                            }
+                        else if(V_j(idx) < 0)
+                            {
+                            // ERROR: can't have V=-inf ever, it is a mass sink
+                            }
+                        }
+                    no_flux = (Vidx_inf || std::isinf(V_j(idx-1)));
+                    }
+                if (!no_flux)
+                    {
+                    const auto x = mesh->lower_bound(idx);
+                    if (x==0)
+                        {
+                        flux_i(idx) = 0;
+                        }
+                    else
+                        {
+                        auto rho_x = mesh->interpolate(x, rho_i);
+                       
+                        //BD flux calculation
+                        if(i==j)
+                            {
+                            auto dmu_ex = 0.0;
+                            if (mu_ex_j) dmu_ex += mesh->gradient(idx,mu_ex_j);
+                            if (V_j) dmu_ex += mesh->gradient(idx,V_j);
+                            double D = 1/(6*M_PI*viscosity_*a_j);
+                            flux_i(idx) += -D*(mesh->gradient(idx,rho_j) + rho_x*dmu_ex);
+                            }
+                       
+                        //RPY flux calculation   
+                        const double d_ij = a_i + a_j;
+                        
+                        //To remove the concern about the lower bound value spill over the buffer sites
+                        double lower_ig = ceil((x-d_ij-mesh->lower_bound())/mesh->step());
+                        const int ig_low = mesh->bin(std::max(lower_ig, 0.));
+                        const int ig_high = floor((mesh->upper_bound()-(x+d_ij))/mesh->step());
+                        double ig = 0.;
+                        for (int ig_idx = ig_low; ig_idx <= ig_high; ++ig_idx)
+                            {
+                            const auto y = mesh->lower_bound(ig_idx);
+                            // total gradient of chemical potential
+                            double rho_dmu = mesh->gradient(ig_idx, rho_j);
+                            auto rho_y = mesh->interpolate(y, rho_j);
+                            if (mu_ex_j)
+                                rho_dmu += rho_y * mesh->gradient(ig_idx, mu_ex_j);
+                            if (V_j)
+                                rho_dmu += rho_y * mesh->gradient(ig_idx, V_j);
+                            // M will need the viscosity too
+                            const double M = ((a_i * a_i + 3 * a_i * a_j + a_j * a_j) * 
+                                        (d_ij - x - y) * (d_ij + x - y) * (d_ij - x + y) 
+                                        * (d_ij + x + y)) / (24 * x * x * viscosity_ * d_ij * d_ij * d_ij);
+                            ig += M * rho_dmu;
+                         
+                            flux_i(idx) += -rho_x * ig * mesh->step();
+                            }
```

Comment:
This needs to go outside the integration loop:
```suggestion
                            }
                        flux_i(idx) += -rho_x * ig * mesh->step();
```
We also still need to check the sign on it.